meerk40t/gui/scene/scenepanel.py

```diff
@@ -70,61 +72,30 @@ def signal(self, *args, **kwargs):
         """
         self.scene._signal_widget(self.scene.widget_root, *args, **kwargs)
 
-    # Indicator for Keyboard-Modifier
-    isShiftPressed = False
-    isCtrlPressed = False
-    isAltPressed = False
+    def event_modifiers(self, event: wx.MouseEvent):
+        modifiers = []
+        if event.RawControlDown() and not event.ControlDown():
+            modifiers.append("macctrl")
+        elif event.ControlDown():
+            modifiers.append("ctrl")
+        if event.AltDown():
+            modifiers.append("alt")
+        if event.ShiftDown():
+            modifiers.append("shift")
+        if event.MetaDown():
+            modifiers.append("meta")
+        return modifiers
 
     def on_key_down(self, evt):
         literal = get_key_name(evt, True)
-        # keycode = evt.GetKeyCode()
-        # print("Key-Down: %f - literal: %s" % (keycode, literal))
-        if "shift+" in literal:
-            if not self.isShiftPressed:  # ignore multiple calls
-                self.isShiftPressed = True
-                self.scene.event(self.scene.last_position, "kb_shift_press", None)
-        if "ctrl+" in literal:
-            if not self.isCtrlPressed:  # ignore multiple calls
-                self.isCtrlPressed = True
-                self.scene.event(self.scene.last_position, "kb_ctrl_press", None)
-        if "alt+" in literal:
-            if not self.isAltPressed:  # ignore multiple calls
-                self.isAltPressed = True
-                self.scene.event(self.scene.last_position, "kb_alt_press", None)
-                # self.scene.event(self.scene.last_position, "kb_alt_press")
+        if literal != self.last_key:
+            self.scene.event(self.scene.last_position, "key_down", None, literal)
+        self.last_key = literal
         evt.Skip()
 
     def on_key_up(self, evt):
-        literal = ""
-        keystates = (evt.ShiftDown(), evt.ControlDown(), evt.AltDown())
-        if self.isShiftPressed and not keystates[0]:
-            literal += "shift+"
-        if self.isCtrlPressed and not keystates[1]:
-            literal += "ctrl+"
-        if self.isAltPressed and not keystates[2]:
-            literal += "alt+"
-        key = evt.GetKeyCode()
-        # print("Key-Up: %f - literal: %s" % (key, literal))
-
-        if "shift+" in literal:
-            if self.isShiftPressed:  # ignore multiple calls
-                self.isShiftPressed = False
-                self.scene.event(self.scene.last_position, "kb_shift_release", None)
-        if "ctrl+" in literal:
-            if self.isCtrlPressed:  # ignore multiple calls
-                self.isCtrlPressed = False
-                self.scene.event(self.scene.last_position, "kb_ctrl_release", None)
-        if "alt+" in literal:
-            if self.isAltPressed:  # ignore multiple calls
-                self.isAltPressed = False
-                self.scene.event(self.scene.last_position, "kb_alt_release", None)
-        if key == 27: # ESC
```

Comment:
This functionality is restored in the tools themselves. See modifier key_up/escape for the tools